transport-native-unix-common/src/main/java/io/netty/channel/unix/NativeInetAddress.java

```diff
@@ -91,7 +90,14 @@ public static InetSocketAddress address(byte[] addr, int offset, int len) {
                     byte[] ipv6 = new byte[16];
                     System.arraycopy(addr, offset, ipv6, 0, 16);
                     int scopeId = decodeInt(addr, offset + len  - 8);
-                    address = Inet6Address.getByAddress(null, ipv6, scopeId);
+                    // Only include the scopeId if its either non 0 or if this is a link-local address
+                    // as scopeId is only supported with it:
+                    // See also https://man7.org/linux/man-pages/man7/ipv6.7.html
+                    if (scopeId != 0 || (ipv6[0] == (byte) 0xfe && ipv6[1] == (byte) 0x80)) {
+                        address = Inet6Address.getByAddress(null, ipv6, scopeId);
+                    } else {
+                        address = InetAddress.getByAddress(null, ipv6);
+                    }
```

Comment:
Reading through the man page, it sounds like scope is _only_ supported for link-local addresses. Does that means we should omit the scope, even non-zero scope, if the address is not link local?

Reply:
I asked myself the same but then decided against it as we should only fill it with non-zero when the underlying OS tells us so..